packages/clay-data-provider/src/useResource.ts

```diff
@@ -19,14 +19,28 @@ import {useCache} from './useCache';
 import {timeout} from './util';
 
 interface IResource extends IDataProvider {
+	/**
+	 * Flag that tells `useResource` not to make the first request as soon as the
+	 * component is mounted, you can do it manually by calling the `get()` method
+	 * returned by the hook.
+	 *
+	 * > const {get: getPosts} = useResource({lazy: true'});
+	 */
+	lazy?: boolean;
+
 	onNetworkStatusChange?: (status: NetworkStatus) => void;
 }
 
+const warningFirstFetchMessage = `DataProvider: The first fetch has already happened, if you want to make a new fetch, use the 'refetch()' method but if you want to make the first fetch enable the 'lazy' hook mode.
```

Comment:
I think we might want to move this to our docs rather than emit a warning. Primarily because they aren't necessarily doing anything wrong, and if they intend to already use the first fetch, they will always see this warning. wdyt?

Reply:
I think it could also be added to the documentation, but this message will only appear in some cases, for example:

- If `lazy` is not enabled, the component once assembled will make the first request and if people calls `get` the message will appear.
- If `lazy` is enabled and people calls `get` twice the message will appear.

The message was not very clear about these cases?

Reply:
This doesn't seem like a very developer-friendly API. As a user, I'd be asking, why should I have two separate methods for "first lazy fetch" versus "fetch again", when they are both methods that "fetch" the data and in both cases I really just mean "go get data now"?

Reply:
Yeah at first yes, but it's weird to have the same `refetch` method that makes the first request and then the subsequent ones when a `lazy` is activated. We are changing your responsibility...

The idea of ​​`lazy` is for cases when you want to execute the query in response to a different event other than the assembly of the component. So we just run it the first time. In essence `refetch` will only be used in other cases and it changes the network status to `Refetch`.

Another approach would be to create a `useLazyResource` that has only this behavior, something similar to React Apollo's `useLazyQuery` but would still have the same idea, a `get` and `refetch`.